helix-tui/src/backend/crossterm.rs

```diff
@@ -153,7 +192,7 @@ impl ModifierDiff {
         if removed.contains(Modifier::ITALIC) {
             map_error(queue!(w, SetAttribute(CAttribute::NoItalic)))?;
         }
-        if removed.contains(Modifier::UNDERLINED) {
+        if removed.intersects(Modifier::ANY_UNDERLINE) {
```

Comment:
Could it be that we have to emit `NoUnderline` followed by underline if the cell changes from one underline to a different type in the next one?